app/controllers/main/routes.py

```diff
@@ -12,30 +13,28 @@ def home():
     print(f"{g.current_user.username}: {g.current_user.firstName} {g.current_user.lastName}")
     return render_template('main/home.html', title="Welcome to CELTS!")
 
-@main_bp.route('/volunteerIndicateInterest', methods = ['GET'])
-def volunteerIndicateInterest():
+@main_bp.route('/profile/<username>', methods = ['GET'])
+def profilePage(username):
+    if username != g.current_user.username or g.current_user.isCeltsAdmin or g.current_user.isCeltsStudentStaff:
+        return "", 500 #TODO: Link to an error page
+    upcomingEvents = getUpcomingEventsForUser(username)
     programs = Program.select()
     interests = Interest.select().where(Interest.user == g.current_user)
     interests_ids = [interest.program for interest in interests]
-    return render_template('volunteerIndicateInterest.html',
-                           title="Volunteer Interest",
-                           user = g.current_user,
-                           programs = programs,
-                           interests = interests,
-                           interests_ids = interests_ids)
-
-
-@main_bp.route('/profile/<username>', methods = ['GET'])
-def profilePage(username):
-
     if username == g.current_user.username or g.current_user.isCeltsAdmin or g.current_user.isCeltsStudentStaff:
-        return f'<h1>Profile page for {username}</h1>'
-
-    abort(403)
+        return render_template('/volunteer/volunteerProfile.html',
+                               title="Volunteer Interest",
+                               user = username,
+                               programs = programs,
+                               interests = interests,
+                               interests_ids = interests_ids,
+                               upcomingEvents = upcomingEvents)
+    else:
+        return "", 500
+        abort(403)
```

Comment:
Nothing after a `return` statement will be executed, so which do you want? A 500 error or a 403 error? Hint: 403 is an access forbidden error.